kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NMSettingsConverterTest.java

```diff
@@ -0,0 +1,641 @@
+		} catch (IllegalArgumentException e) {
+			e.printStackTrace();
+			hasIllegalArgumentExceptionBeenThrown = true;
+		} catch (Exception e) {
+			e.printStackTrace();
+			hasAGenericExecptionBeenThrown = true;
+		}
+	}
+
+	public void whenBuildIpv6SettingsIsRunWith(NetworkProperties props, String iface) {
+		try {
+			this.resultMap = NMSettingsConverter.buildIpv6Settings(props, iface);
+		} catch (IllegalArgumentException e) {
+			e.printStackTrace();
+			hasIllegalArgumentExceptionBeenThrown = true;
+		} catch (Exception e) {
+			e.printStackTrace();
+			hasAGenericExecptionBeenThrown = true;
+		}
+	}
+
+	public void whenBuild80211WirelessSettingsIsRunWith(NetworkProperties props, String iface) {
+		try {
+			this.resultMap = NMSettingsConverter.build80211WirelessSettings(props, iface);
+		} catch (IllegalArgumentException e) {
+			e.printStackTrace();
+			hasIllegalArgumentExceptionBeenThrown = true;
+		} catch (Exception e) {
+			e.printStackTrace();
+			hasAGenericExecptionBeenThrown = true;
+		}
+	}
+
+	public void whenBuild80211WirelessSecuritySettingsIsRunWith(NetworkProperties props, String iface) {
+		try {
+			this.resultMap = NMSettingsConverter.build80211WirelessSecuritySettings(props, iface);
+		} catch (IllegalArgumentException e) {
+			e.printStackTrace();
+			hasIllegalArgumentExceptionBeenThrown = true;
+		} catch (Exception e) {
+			e.printStackTrace();
+			hasAGenericExecptionBeenThrown = true;
+		}
+	}
+
+	// then
+	public void thenResultingMapContains(String key, Object value) {
+		assertEquals(value, this.resultMap.get(key).getValue());
+	}
+	
+	public void thenResultingMapContainsBytes(String key, Object value) {
+		assertEquals(value, new String((byte[]) this.resultMap.get(key).getValue(), StandardCharsets.UTF_8));
+	}
+
+	public void thenResultingBuildAllMapContains(String key, String subKey, Object value) {
+		assertEquals(value, this.resultAllSettingsMap.get(key).get(subKey).getValue());
+	}
+	
+	public void thenResultingBuildAllMapContainsBytes(String key, String subKey, Object value) {
+		assertEquals(value, new String((byte[]) this.resultAllSettingsMap.get(key).get(subKey).getValue(), StandardCharsets.UTF_8));
+	}
+
+	public void thenIllegalArgumentExceptionHasBeenThrown() {
+		assertTrue(this.hasIllegalArgumentExceptionBeenThrown);
+	}
+
+	public void thenNoExceptionsHaveBeenThrown() {
+		assertFalse(this.hasIllegalArgumentExceptionBeenThrown);
+		assertFalse(this.hasAGenericExecptionBeenThrown);
+	}
+
+	//helper classes
+	public Object buildAddressDataWith(String ipAddr, UInt32 prefix) {
+		
+		Map<String, Variant<?>> addressEntry = new HashMap<>();
+		addressEntry.put("address", new Variant<>(ipAddr));
+		addressEntry.put("prefix", new Variant<>(prefix));
+		
+		List<Map<String, Variant<?>>> addressData = Arrays.asList(addressEntry);
+		
+		Variant<?> dataVarient = new Variant<>(addressData, "aa{sv}");
+		
+		return dataVarient.getValue();
+	}
```

Comment:
Typo on `Varient`, should be `Variant`:

```suggestion
	public Object buildAddressDataWith(String ipAddr, UInt32 prefix) {
		
		Map<String, Variant<?>> addressEntry = new HashMap<>();
		addressEntry.put("address", new Variant<>(ipAddr));
		addressEntry.put("prefix", new Variant<>(prefix));
		
		List<Map<String, Variant<?>>> addressData = Arrays.asList(addressEntry);
		
		Variant<?> dataVariant = new Variant<>(addressData, "aa{sv}");
		
		return dataVariant.getValue();
	}
```